types/components/buttons.d.ts

```diff
@@ -108,6 +108,81 @@ export type OnShippingChangeActions = {
     };
 };
 
+type CurrencyCodeAndValue = {
+    currencyCode: string;
+    value: string;
+};
+
+type ShippingOption = {
+    amount: CurrencyCodeAndValue;
+    id?: string;
+    label: string;
+    selected: boolean;
+    type: "SHIPPING" | "PICKUP";
+};
+
+export type OnShippingOptionsChangeData = {
+    orderID?: string;
+    paymentID?: string;
+    paymentToken?: string;
+    selectedShippingOption?: ShippingOption;
+};
+
+type BuildOrderPatchPayloadOptions = {
+    discount?: string;
+    handling?: string;
+    insurance?: string;
+    itemTotal?: string;
+    option?: ShippingOption;
+    shippingOptions?: ShippingOption[];
+    shippingDiscount?: string;
+    taxTotal?: string;
+};
+
+export type OnShippingOptionsChangeActions = {
+    buildOrderPatchPayload: ({
+        discount,
+        handling,
+        insurance,
+        itemTotal,
+        shippingOptions,
+        shippingDiscount,
+        taxTotal,
+    }: BuildOrderPatchPayloadOptions) => UpdateOrderRequestBody;
+    reject: () => Promise<void>;
+};
+
+export type OnShippingAddressChangeData = {
+    amount: CurrencyCodeAndValue;
+    orderID?: string;
+    paymentID?: string;
+    paymentToken?: string;
+    shippingAddress: {
+        city: string;
+        state: string;
+        /** The [two-character ISO 3166-1 code](/docs/integration/direct/rest/country-codes/) that identifies the country or region. */
+        countryCode: string;
+        /**
+         * The postal code, which is the zip code or equivalent.
+         * Typically required for countries with a postal code or an equivalent.
+         */
+        postalCode: string;
+    };
+};
+
+export type OnShippingAddressChangeActions = {
+    buildOrderPatchPayload: ({
+        discount,
+        handling,
+        insurance,
+        itemTotal,
+        shippingOptions,
+        shippingDiscount,
+        taxTotal,
+    }: BuildOrderPatchPayloadOptions) => UpdateOrderRequestBody;
+    reject: () => Promise<void>;
+};
```

Comment:
I don't think these types need to be exported if they are not imported from anywhere else. I'd lean toward removing the `export` in this case..

Reply:
I'm with you on that, though I left at least one `export` in SPB even though it wasn't being imported anywhere. 😄